### PR TITLE
refactor: replace `build-adapters` feature by `docs`

### PR DESCRIPTION
**Pull request by rvolosatovs**
## Feature or Problem
 

Skipping the adapter fetch is only useful for documentation purposes, introduce `docs` feature specifically for this purpose, such that this crate could still be used with of without default features enabled downstream

## Related Issues

#10 
#11 

## Testing

Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

**Comment by connorsmith256** [comment]
This is fine, but just know you'll need to update the feature flag transitively in wash, like I did here https://github.com/wasmCloud/wash/pull/864

File: build.rs [hunk 1]
```diff
@@ -172,11 +172,10 @@ async fn main() -> anyhow::Result<()> {
...
 
-    #[cfg(feature = "build-adapters")]
-    {
-        let out_dir = env::var("OUT_DIR")
-            .map(PathBuf::from)
-            .context("failed to lookup `OUT_DIR`")?;
+    let out_dir = env::var("OUT_DIR")
+        .map(PathBuf::from)
+        .context("failed to lookup `OUT_DIR`")?;
+    if cfg!(feature = "docs") {
```
Comment (connorsmith256) [inline build.rs:178]: Should this condition/the blocks be inverted?

Comment (rvolosatovs) [reply build.rs:178]: yes :eyes:

**Review by connorsmith256 — APPROVED** [review]
LGTM. We'll want a version bump to release 👍

**Pull request merged**